x-pack/plugins/ml/public/application/datavisualizer/stats_datagrid/stats_datagrid.tsx

```diff
@@ -20,60 +20,57 @@ import {
 import { i18n } from '@kbn/i18n';
 import { EuiTableComputedColumnType } from '@elastic/eui/src/components/basic_table/table_types';
 import { FieldTypeIcon } from '../../components/field_type_icon';
-import { FieldVisConfig } from '../index_based/common';
-import { DataVisualizerFieldExpandedRow } from './expanded_row';
 import { DocumentStat } from './components/field_data_row/document_stats';
 import { DistinctValues } from './components/field_data_row/distinct_values';
 import { NumberContentPreview } from './components/field_data_row/number_content_preview';
-import { DataVisualizerIndexBasedAppState } from '../../../../common/types/ml_url_generator';
+import {
+  DataVisualizerFileBasedAppState,
+  DataVisualizerIndexBasedAppState,
+} from '../../../../common/types/ml_url_generator';
 import { useTableSettings } from '../../data_frame_analytics/pages/analytics_management/components/analytics_list/use_table_settings';
 import { TopValuesPreview } from './components/field_data_row/top_values_preview';
 import type { MlJobFieldType } from '../../../../common/types/field_types';
+import {
+  FieldVisConfig,
+  FileBasedFieldVisConfig,
+  isIndexBasedFieldVisConfig,
+} from './types/field_vis_config';
 const FIELD_NAME = 'fieldName';
 
-interface DataVisualizerDataGrid {
-  items: FieldVisConfig[];
-  pageState: DataVisualizerIndexBasedAppState;
-  updatePageState: (update: Partial<DataVisualizerIndexBasedAppState>) => void;
-}
-
 export type ItemIdToExpandedRowMap = Record<string, JSX.Element>;
 
-function getItemIdToExpandedRowMap(
-  itemIds: string[],
-  items: FieldVisConfig[]
-): ItemIdToExpandedRowMap {
-  return itemIds.reduce((m: ItemIdToExpandedRowMap, fieldName: string) => {
-    const item = items.find((fieldVisConfig) => fieldVisConfig[FIELD_NAME] === fieldName);
-    if (item !== undefined) {
-      m[fieldName] = <DataVisualizerFieldExpandedRow item={item} />;
-    }
-    return m;
-  }, {} as ItemIdToExpandedRowMap);
+type DataVisualizerDataGridItem = FieldVisConfig | FileBasedFieldVisConfig;
+interface DataVisualizerDataGridProps<T> {
+  items: T[];
+  pageState: DataVisualizerIndexBasedAppState | DataVisualizerFileBasedAppState;
+  updatePageState: (update: any) => void;
+  getItemIdToExpandedRowMap: (itemIds: string[], items: T[]) => ItemIdToExpandedRowMap;
 }
 
-export const DataVisualizerDataGrid = ({
+export const DataVisualizerDataGrid = <T extends DataVisualizerDataGridItem>({
```

Comment:
Nit - but I think this folder should be renamed `stats_data_grid`, or even just `data_grid` ?

Reply:
the word "grid" isn't really relevant anymore as this PR turns it into a table.

Reply:
Renamed the word DataGrid to `Table` here https://github.com/elastic/kibana/pull/87598/commits/ea57967e89c25c03a7de83aec9f802caf78c1708

Reply:
the file names are still called "grid"